lumibot/tools/polygon_helper.py

```diff
@@ -411,6 +411,20 @@ def get_missing_dates(df_all, asset, start, end):
     dates = pd.Series(df_all.index.date).unique()
     missing_dates = sorted(set(trading_dates) - set(dates))
 
+    # TODO: This code works AFAIK, But when i enable it the tests for "test_polygon_missing_day_caching" and
+    # i don't know why nor how to fix this code or the tests. So im leaving it disabled for now. If you have problems
+    # with NANs in cached polygon data, you can try to enable this code and fix the tests.
+
+    # # Find any dates with nan values in the df_all DataFrame
+    # missing_dates += df_all[df_all.isnull().all(axis=1)].index.date.tolist()
```

Comment:
I believe this is the line that doesn't work for you.  Completely missing dates are not guarenteed to have an all-empty entry in the DataFrame.  This commented out section can be fully removed.